### PR TITLE
🧹  Updating AWS Policies to Align with Recent Developments

### PR DESCRIPTION
**Pull request by HRouhani**

### CODE REVIEW

File: core/mondoo-aws-security.mql.yaml [hunk 1]
```diff
@@ -1866,28 +1936,41 @@ queries:
     filters: |
       asset.platform == "aws-lambda-function"
     mql: |
-      aws.lambda.function.concurrency > 0
+      aws.lambda.function.concurrency <= 100
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:1939]: If we do not use properties, I suggest keeping the `> 0` check in addition to the `<=100` check, as sensible values.
```suggestion
      aws.lambda.function.concurrency > 0
      aws.lambda.function.concurrency <= 100
```

File: core/mondoo-aws-security.mql.yaml [hunk 2]
```diff
@@ -1866,28 +1936,41 @@ queries:
...
       asset.platform == "aws"
     mql: |
-      aws.lambda.functions.all(concurrency > 0)
-  - uid: mondoo-aws-security-rds-instance-public-access-check
-    title: Ensure all RDS instances are not publicly accessible
-    impact: 100
-    variants:
-      - uid: mondoo-aws-security-rds-instance-public-access-check-all
-      - uid: mondoo-aws-security-rds-instance-public-access-check-single
+      aws.lambda.functions.all(concurrency <= 100)
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:1944]: same as above

File: core/mondoo-aws-security.mql.yaml [hunk 3]
```diff
@@ -1994,7 +2077,8 @@ queries:
     title: Ensure Redshift clusters are not publicly accessible
     impact: 95
     filters: asset.platform == "aws-redshift-cluster"
-    mql: aws.redshift.cluster.publiclyAccessible == false
+    mql: |
+      aws.redshift.clusters.all(publiclyAccessible == false)
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:2081]: The check is on the `filters: asset.platform == "aws-redshift-cluster"`, which means we need to keep this syntax:

```suggestion
    mql: |
      aws.redshift.cluster.publiclyAccessible == false
```

File: core/mondoo-aws-security.mql.yaml [hunk 4]
```diff
@@ -2065,11 +2149,14 @@ queries:
         title: Terraform Documentation - AWS Provider
   - uid: mondoo-aws-security-ec2-volume-inuse-check
     title: Ensure EBS volumes attached to EC2 instances are configured for deletion on instance termination
-    impact: 30
-    filters: |
-      asset.platform == "aws-ec2-volume"
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:2070]: This check was also originally on ` asset.platform == "aws-ec2-volume"`, so we should keep the single instance check below, too.

File: core/mondoo-aws-security.mql.yaml [hunk 5]
```diff
@@ -2142,19 +2229,17 @@ queries:
   - uid: mondoo-aws-security-ebs-snapshot-public-restorable-check
     title: Ensure EBS snapshots are not publicly restorable
     impact: 80
-    filters: |
-      asset.platform == "aws-ec2-snapshot" || asset.platform == "aws-ebs-snapshot"
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:2146]: same as above, some customers might be using this on this specific asset, there's no need to change the asset type.

**Review by mm-weber — COMMENTED** [review]
Generally it's a good idea to update the checks to keep in line with the new developments, but there's a problem with changing the `asset_filters:` of existing checks to other assets.

Many checks have been moved from the single asset, to the cluster asset, which 1) isn't the direction we're moving with checks in general and 2) more importantly, existing customers might have already worked with the single asset checks. 

Would be great if we can keep the asset_filter scopes and just update the checks as needed.

File: core/mondoo-aws-security.mql.yaml (hunk 3 above)
Comment (HRouhani) [reply core/mondoo-aws-security.mql.yaml]: Thank you. I've already deployed a variant version for this test. Meanwhile, investigations are ongoing regarding the presence of a Redshift cluster in the new AWS development environment. If it's confirmed that we're transitioning entirely to serverless, I'll remove the associated checks.

File: core/mondoo-aws-security.mql.yaml (hunk 5 above)
Comment (HRouhani) [reply core/mondoo-aws-security.mql.yaml:2146]: changed to variant, done

File: core/mondoo-aws-security.mql.yaml [hunk 6]
```diff
@@ -81,54 +82,69 @@ policies:
           - uid: mondoo-aws-security-iam-user-no-inline-policies-check
           - uid: mondoo-aws-security-iam-user-no-inline-policies-check-single-user
       - title: AWS Lambda Function
+        filters: asset.platform == "aws"
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:85]:
```suggestion
```

Comment (mm-weber) [reply core/mondoo-aws-security.mql.yaml:85]: With the `variants:` checks we don't need `filters:` e.g. ` mondoo-aws-security-lambda-concurrency-check-single` would never fire with the group filter in place.

File: core/mondoo-aws-security.mql.yaml [hunk 7]
```diff
@@ -81,54 +82,69 @@ policies:
...
         checks:
           - uid: mondoo-aws-security-lambda-concurrency-check
       - title: AWS S3 Bucket
+        filters: asset.platform == "aws"
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:89]: Same here

File: core/mondoo-aws-security.mql.yaml [hunk 8]
```diff
@@ -70,6 +70,7 @@ policies:
         If you have any suggestions for how to improve this policy or need support, [join the community](https://github.com/orgs/mondoohq/discussions) in GitHub Discussions.
     groups:
       - title: AWS IAM
+        filters: asset.platform == "aws"
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:73]:
```suggestion
```

**Review by mm-weber — CHANGES_REQUESTED** [review]
Thanks for the changes to the checks @HRouhani . Those all look good.

We need to double check and most likely remove all ` filters: asset.platform == "aws"` at group level as those will never allow any of the single asset level `variants:` to run.

**Comment by HRouhani** [comment]
@mm-weber It should be fine that. I did it exactly what what we did in Azure, using asset.runtime. You can test all variant now using following script:

```
#!/bin/bash

# List of AWS resource types to check
resource_types=(
  "ssm-instances"
  "ssm-instances-api"
  "ecr"
  "ecs"
  "s3-buckets"
  "cloudtrail-trails"
  "rds-dbinstances"
  "rds-dbclusters"
  "vpcs"
  "security-groups"
  "iam-users"
  "iam-groups"
  "cloudwatch-loggroups"
  "lambda-functions"
  "dynamodb-tables"
  "redshift-clusters"
  "ec2-volumes"
  "ec2-snapshots"
  "efs-filesystems"
  "gateway-restapis"
  "elb-loadbalancers"
  "es-domains"
  "kms-keys"
  "sagemaker-notebookinstances"
  "accounts"
  "ec2-instances-api"
  "instances"
)

# Path to the cnspec binary and the MQL file
cnspec_path="/home/.../cnspec"
bundle="/home/....mondoo-aws-security.mql.yaml"

# Loop over each resource type and execute the scan
for resource in "${resource_types[@]}"; do
  echo "********Discovering the $resource  resources and Starting to scan it respectively.....*********"
  $cnspec_path scan aws -f $bundle --discover "$resource"
done

echo "Scanning completed."
```  

Change the cnspec path to your binary as well as the bundle  path.

**Review by mm-weber — APPROVED** [review]
Good stuff @HRouhani !
Thank you.

**Pull request merged**